src/addrman.h

```diff
@@ -176,6 +177,20 @@ friend class CAddrManTest;
     mutable RecursiveMutex cs;
 
 private:
+    //! Serialization versions.
+    enum Format : uint8_t {
+        V0_HISTORICAL = 0,    //!< historic format, before commit e6b343d88
+        V1_DETERMINISTIC = 1, //!< for pre-asmap files
+        V2_ASMAP = 2,         //!< for files including asmap version
+        V3_BIP155 = 3,        //!< same as V2_ASMAP plus addresses are in BIP155 format
+    };
+
+    //! The initial value of a field that is incremented every time an incompatible format
+    //! change is made. This is 32 instead of 0 because we overtook the "key size" field
+    //! which was 32 historically and we don't want to start with 0 and bump into 32 at
+    //! some point in the future.
+    static constexpr uint8_t m_incompatibility_base = 32;
```

Comment:
nit: Any value equal to or greater than 32 should be fine here. As this is a breaking change (old versions can't read the peers.dat created by this version of Bitcoin Core), might as well bump to 33 right away to avoid confusion with the "magic value" 32.

Reply:
Or perhaps use `0x80` as a very obvious offset. We're unlikely to have 128 file versions so we're not going to run out of bits.

Reply:
I think it should be one of `29`, `30`, `31` or `32`:

```cpp
// Unserialize in 0.21
lowest_compatible = byte_read_from_disk - INCOMPATIBILITY_BASE;
```

0.20 write `32` to disk

So

* If `INCOMPATIBILITY_BASE` is `28` or less then a file written by 0.20 will be rendered unreadable by 0.21 as having too high `lowest_compatible`: `4` or higher.
* If `INCOMPATIBILITY_BASE` is one of `29`, `30`, `31` or `32` then a file written by 0.20 will appear to have `lowest_compatible` `3`, `2`, `1` or `0` which is ok for 0.21.
* If `INCOMPATIBILITY_BASE` is `33` or greater then a file written by 0.20 will cause 0.21 to calculate a negative `lowest_compatible`.

So I left it as `32`.